core/src/consensus.rs

```diff
@@ -646,7 +646,9 @@ impl Tower {
                     // By the time we reach here, any ancestors of the `last_vote`,
                     // should have been filtered out, as they all have a descendant,
                     // namely the `last_vote` itself.
-                    assert!(!last_vote_ancestors.contains(candidate_slot));
+                    if !self.is_stray_last_vote() {
```

Comment:
Investigating why this is necessary, hopefully just a test issue 🙏

Reply:
@ryoqun, I commented this check out,  but haven't been able to repro the failure you described. The test seems to be passing fine on several different machines.